### PR TITLE
Added logic to NotebookImportExtractor to guess the encoding on initi…

### PR DESCRIPTION
**Pull request by fpgmaas**
…al UnicodeDecodeError

**PR Checklist**

-   [x] A description of the changes is added to the description of this PR.
-   [x] If there is a related issue, make sure it is linked to this PR.
-   [x] If you've fixed a bug or added code that should be tested, add tests!
-   [ ] Documentation in `docs` is updated

**Description of changes**

### CODE REVIEW

File: deptry/imports/extractors/notebook_import_extractor.py [hunk 1]
```diff
@@ -17,17 +18,24 @@ class NotebookImportExtractor(ImportExtractor):
...
-    @staticmethod
-    def _read_ipynb_file(path_to_ipynb: Path) -> dict[str, Any]:
-        with open(path_to_ipynb) as f:
-            notebook: dict[str, Any] = json.load(f)
+        if notebook:
+            cells = self._keep_code_cells(notebook)
+            import_statements = [self._extract_import_statements_from_cell(cell) for cell in cells]
+            tree = ast.parse("\n".join(itertools.chain.from_iterable(import_statements)), str(self.file))
+            return self._extract_imports_from_ast(tree)
+        return set()
```
Comment (mkniewallner) [inline deptry/imports/extractors/notebook_import_extractor.py:26]:
```suggestion
        if not notebook:
            return set()
        
        cells = self._keep_code_cells(notebook)
        import_statements = [self._extract_import_statements_from_cell(cell) for cell in cells]
        tree = ast.parse("\n".join(itertools.chain.from_iterable(import_statements)), str(self.file))
        return self._extract_imports_from_ast(tree)
```

File: deptry/imports/extractors/notebook_import_extractor.py [hunk 2]
```diff
@@ -17,17 +18,24 @@ class NotebookImportExtractor(ImportExtractor):
...
+
+    def _read_ipynb_file(self, path_to_ipynb: Path) -> dict[str, Any] | None:
+        try:
+            with open(path_to_ipynb) as ipynb_file:
+                notebook: dict[str, Any] = json.load(ipynb_file)
+        except UnicodeDecodeError:
+            try:
+                with open(path_to_ipynb, encoding=self._get_file_encoding(path_to_ipynb)) as ipynb_file:
```
Comment (mkniewallner) [inline deptry/imports/extractors/notebook_import_extractor.py:34]:
```suggestion
    @classmethod
    def _read_ipynb_file(cls, path_to_ipynb: Path) -> dict[str, Any] | None:
        try:
            with open(path_to_ipynb) as ipynb_file:
                notebook: dict[str, Any] = json.load(ipynb_file)
        except UnicodeDecodeError:
            try:
                with open(path_to_ipynb, encoding=cls._get_file_encoding(path_to_ipynb)) as ipynb_file:
```

**Review by mkniewallner — APPROVED** [review]
Some nits, but non-blockers, so approving.

**Comment by fpgmaas** [comment]
> Some nits, but non-blockers, so approving.

Thanks for reviewing!

**Pull request merged**